src/main/java/mil/dds/anet/resources/PersonResource.java

```diff
@@ -196,6 +196,25 @@ public Integer updatePerson(@GraphQLRootContext Map<String, Object> context,
     return numRows;
   }
 
+  @GraphQLMutation(name = "updatePersonHistory")
+  public int updatePersonHistory(@GraphQLRootContext Map<String, Object> context,
+      @GraphQLArgument(name = "person") Person p) {
+    final Person user = DaoUtils.getUserFromContext(context);
+    final Person existing = dao.getByUuid(p.getUuid());
+    if (!canCreateOrUpdatePerson(user, existing, false)) {
+      throw new WebApplicationException("You do not have permissions to edit this person",
+          Status.FORBIDDEN);
+    }
```

Comment:
This can be extracted into `assertCanUpdatePerson(user, existing, false)` (also in other places this is used).